src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -155,8 +144,6 @@ public void CompleteStream(bool errored)
                 // The app can no longer read any more of the request body, so return any bytes that weren't read to the
                 // connection's flow-control window.
                 _inputFlowControl.Abort();
-
-                Reset();
```

Comment:
It's too bad we lost this. Resetting when the stream completes is better than resetting when we initialize, because then we don't root headers and the like when the stream is in the pool.